graylog2-server/src/main/java/org/graylog2/migrations/V20170607164210_MigrateReopenedIndicesToAliases.java

```diff
@@ -0,0 +1,138 @@
+        this.jestClient = jestClient;
+    }
+
+    @Override
+    public ZonedDateTime createdAt() {
+        return ZonedDateTime.parse("2017-06-07T16:42:10Z");
+    }
+
+    // Create aliases for legacy reopened indices.
+    @Override
+    public void upgrade() {
+        this.indexSetService.findAll()
+            .stream()
+            .map(mongoIndexSetFactory::create)
+            .flatMap(indexSet -> getReopenedIndices(indexSet).stream())
+            .map(indexName -> { LOG.debug("Marking index {} to be reopened using alias.", indexName); return indexName; })
+            .forEach(indices::markIndexReopened);
+    }
+
+    private Set<String> getReopenedIndices(final Collection<String> indices) {
+        final Version elasticsearchVersion = node.getVersion().orElseThrow(() -> new ElasticsearchException("Unable to retrieve Elasticsearch version."));
+        final String indexList = String.join(",", indices);
+        final State request = new State.Builder().withMetadata().indices(indexList).build();
+
+        final JestResult jestResult = JestUtils.execute(jestClient, request, () -> "Couldn't read cluster state for reopened indices " + indices);
+        final JsonNode clusterStateJson = jestResult.getJsonObject();
+        final JsonNode indicesJson = clusterStateJson.path("metadata").path("indices");
+
+        final ImmutableSet.Builder<String> reopenedIndices = ImmutableSet.builder();
+
+        if (indicesJson.isMissingNode()) {
+            LOG.error("Retrieved cluster state is invalid (no metadata.indices key).");
+            LOG.debug("Received cluster state was: {}", clusterStateJson.toString());
+            return Collections.emptySet();
+        }
+
+        for (Iterator<Map.Entry<String, JsonNode>> it = indicesJson.fields(); it.hasNext(); ) {
+            final Map.Entry<String, JsonNode> entry = it.next();
+            final String indexName = entry.getKey();
+            final JsonNode value = entry.getValue();
+
+            final JsonNode indexSettings = value.path("settings").path("index");
+            if (indexSettings.isMissingNode()) {
+                LOG.error("Unable to retrieve index settings from metadata for index {} - skipping.", indexName);
+                LOG.debug("Index metadata was: {}", value.toString());
+                continue;
+            }
+            if (checkForReopened(indexSettings, elasticsearchVersion)) {
+                LOG.debug("Adding {} to list of indices to be migrated.", indexName);
+                reopenedIndices.add(indexName);
+            }
+        }
+
+        return reopenedIndices.build();
+    }
+
+    private boolean checkForReopened(@Nullable JsonNode indexSettings, Version elasticsearchVersion) {
+        final JsonNode settings;
+        if (elasticsearchVersion.satisfies(">=2.1.0 & <5.0.0")) {
+            settings = indexSettings;
+        } else if (elasticsearchVersion.satisfies("^5.0.0")) {
+            settings = indexSettings.path("archived");
```

Comment:
As per the comment above, in case of upgrading to 5.x, the `indexSettings` here does not contain the `archived` subkey anymore, so the check comes back as `false`.